src/common/ant.ts

```diff
@@ -278,6 +280,85 @@ export class AoANTReadable implements AoANTRead {
     if (strict) parseSchemaResult(z.number(), balance);
     return balance;
   }
+
+  /**
+   * @returns {Promise<AoANTHandler[]>} The handlers of the ANT.
+   * @example
+   * Get the handlers of the ANT.
+   * ```ts
+   * const handlers = await ant.getHandlers();
+   * ```
+   */
+  async getHandlers(): Promise<AoANTHandler[]> {
+    const info = await this.getInfo();
+
+    return (info.Handlers ?? info.HandlerNames) as AoANTHandler[];
+  }
+
+  /**
+   * @param validations @type {Partial<Record<AoANTHandler, ({ ant: AoANTReadable }) => Promise<boolean>>} The validations to run on the ANT. 
+   * @returns @type {Promise<Record<AoANTHandler, { valid: boolean; error?: string }>>} The results of the validations.
+   * @example
+   * ```ts
+   * const validity = await ant.validate({
+   * validations: {
+   *  'setTicker': async ({ant}: { ant: AoANTRead }) => {
+   *   const antHandlers = await ant.getHandlers();
+            if (antHandlers.includes("setTicker")) return true;
+            throw new Error(`Handler setTicker not found`);
+          }
+    
+   * },
+          //  ...rest of the ANT handlers config
+   * });
+   * ```
+   */
+  async validate({
+    validations,
+  }: {
+    validations?: Partial<
+      Record<
+        AoANTHandler,
+        ({ ant }: { ant: AoANTReadable }) => Promise<boolean>
```

Comment:
if this is not static, why do I need to provide `ant`? shouldn't I be using `this`?. 

how you have it, I could create an instance of an ANT and then use it to validate ANOTHER instance. That seems odd. Either make it static or use `this`.

Reply:
https://github.com/ar-io/ar-io-sdk/pull/237#discussion_r1837018667

Reply:
if we didn't hav ethis function, and only exposed `getHandlers` on the ANT contract - how would we do validation on the arns.app?

I'm considering the idea of just removing this, and letting clients decide on how they want to validate an ANT based on the handlers available.